app/services/support_interface/ministerial_report_candidates_export.rb

```diff
@@ -0,0 +1,118 @@
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:offer]
+      elsif choice_statuses.any? { |choice_status| %w[awaiting_provider_decision interviewing].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:awaiting_provider_decision]
+      elsif choice_statuses.any? { |choice_status| %w[offer_withdrawn].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:offer_withdrawn]
+      elsif choice_statuses.any? { |choice_status| %w[cancelled declined].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:declined]
+      elsif choice_statuses.any? { |choice_status| %w[rejected].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:rejected]
+      elsif choice_statuses.any? { |choice_status| %w[withdrawn].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:withdrawn]
+      end
+    end
+
+  private
+
+    def add_row_values(hash, subject, state)
+      hash[:stem][state] += 1 if MinisterialReport::STEM_SUBJECTS.include? subject
+      hash[:ebacc][state] += 1 if MinisterialReport::EBACC_SUBJECTS.include? subject
+      hash[:secondary][state] += 1 if MinisterialReport::SECONDARY_SUBJECTS.include? subject
+      hash[subject][state] += 1
+    end
+
+    def determine_subjects(application_form)
+      courses = application_form.application_choices.map(&:course)
+      courses.map { |course| dominant_subject_for_course(course) }
+    end
+
+    def dominant_subject_for_course(course)
+      dominant_subject = if course.subjects.size == 1
+                           course.subjects.first
+                         else
+                           main_subject_name = course.name.split[0]
+                           course.subjects.find { |subject| main_subject_name.downcase.in?(subject.name.downcase) }
+                         end
+      MinisterialReport::SUBJECT_CODE_MAPPINGS[dominant_subject.code]
+    end
+
+    def candidate_has_no_dominant_subject?(mapped_subjects)
+      return false if mapped_subjects.count == 1 || mapped_subjects.uniq.size == 1
+
+      return true if count_of_subject_choices(mapped_subjects).values.uniq.size == 1
+    end
+
+    def count_of_subject_choices(subjects)
+      subjects.tally
+    end
+
+    def dominant_subject(subjects)
+      count_of_subject_choices(subjects).max_by(&:last).first
+    end
+
+    def column_names
+      {
+        candidates: 0,
+        candidates_holding_offers: 0,
+        candidates_that_have_accepted_offers: 0,
+        declined_candidates: 0,
+        rejected_candidates: 0,
+        candidates_that_have_withdrawn_offers: 0,
+      }
+    end
+
+    def application_forms
+      ApplicationForm
+        .joins(application_choices: { course: :subjects })
+        .current_cycle.where(phase: 'apply_1')
+        .where.not(application_choices: { id: nil })
```

Comment:
I don't think this line is needed because the INNER JOIN above takes care of the condition